tactics/tactics.ml

```diff
@@ -5323,6 +5323,26 @@ let unify ?(state=TransparentState.full) x y =
     Proofview.tclZERO ~info (PretypeError (env, sigma, CannotUnify (x, y, None)))
   end
 
+let evarconv_unify ?flags ?(with_ho=true) x y =
+  Proofview.Goal.enter begin fun gl ->
+  let env = Proofview.Goal.env gl in
+  let sigma = Proofview.Goal.sigma gl in
+  try
+    let flags =
+      match flags with
+      | Some flags -> flags
+      | None ->
+          (* Pick a more sensible default than [Evarconv.unify] itself. *)
```

Comment:
Is this really a good idea? Seems like we would get a lot of confusion from it

Reply:
The idea was to support `with_strategy`, which seems like a good idea in general, though other unification tactics don't seem to support it. The [documentation of Strategy](https://coq.inria.fr/refman/proofs/writing-proofs/equality.html#coq:cmd.Strategy) does mention "all tactics", so that seems like a bug to me that they are not.

This is also necessary to provide a test case for https://github.com/coq/coq/pull/17788.

Reply:
Having the _option_ of following `with_strategy` sounds good, but wouldn't this differ from `apply:`/`refine` etc? That sounds bad, but could you fix `evarconv` itself? At least math-comp avoids `Opaque` entirely so it shouldn't be affected.

Reply:
Maybe the transparent state should be manipulated explicitly. We could have 
~~~coq
Ltac2 Type transparent_state.

Ltac2 @ external full_transparent_state : transparent_state := ...

Ltac2 @ external current_transparent_state : unit -> transparent_state := ...

Ltac2 @ external evarconv : transparent_state -> constr -> constr -> unit
~~~

Reply:
Yeah, we did considered something like that. And this is probably the best option.

If we go this route, should we make `unify` use `Evarconv.unify` with full transparency state, and additionally add a new `unify_with_params` function with the signature you propose?

Reply:
Now that I think about it, how about we implement the interface you proposed as a new `Unification` module in the Ltac2 standard library? The `Std` module is getting quite crowded, and that would allow us to keep the `unify` tactic as it is in `Std`. We could then try to move `Std.unify` and Ltac's `unify` to `Evarconv` in a second step.

Reply:
Used a specific modules (`TransparentState` and `Unification`).